pkg/_internal/policy/policy_test.go

```diff
@@ -5,60 +5,73 @@ package policy
-		policy := &v1alpha1.DNSPolicy{
-			TypeMeta: metav1.TypeMeta{},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "test-policy",
-				Namespace: "test-policy-ns",
-			},
-			Spec: v1alpha1.DNSPolicySpec{
-				TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
-					Group: "gateway.networking.k8s.io",
-					Kind:  "Gateway",
-					Name:  gatewayv1beta1.ObjectName("test-gateway"),
-				},
-			},
-		}
-
-		res := GetTargetRefValueFromPolicy(policy)
-		expected := "test-gateway,test-policy-ns"
-		if res != expected {
-			t.Errorf("GetTargetRefValueFromPolicy returned %v; expected %v", res, expected)
-		}
-	})
+	type args struct {
+		policy common.KuadrantPolicy
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "should use target namespace",
+			args: args{
+				policy: &v1alpha1.DNSPolicy{
 
-	t.Run("should use policy namespace when no target namespace set", func(t *testing.T) {
-		typedNamespace := gatewayv1beta1.Namespace("test-gateway-ns")
-		policy := &v1alpha1.DNSPolicy{
-			TypeMeta: metav1.TypeMeta{},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "test-policy",
-				Namespace: "test-policy-ns",
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-policy",
+						Namespace: "test-policy-ns",
+					},
+					Spec: v1alpha1.DNSPolicySpec{
+						TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
+							Group: "gateway.networking.k8s.io",
+							Kind:  "Gateway",
+							Name:  "test-gateway",
+						},
+					},
+				},
 			},
-			Spec: v1alpha1.DNSPolicySpec{
-				TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
-					Group:     "gateway.networking.k8s.io",
-					Kind:      "Gateway",
-					Name:      gatewayv1beta1.ObjectName("test-gateway"),
-					Namespace: &typedNamespace,
+			want: "test-gateway,test-policy-ns",
+		},
+		{
+			name: "should use policy namespace when no target namespace set",
+			args: args{
+				policy: &v1alpha1.DNSPolicy{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-policy",
+						Namespace: "test-policy-ns",
+					},
+					Spec: v1alpha1.DNSPolicySpec{
+						TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
+							Group:     "gateway.networking.k8s.io",
+							Kind:      "Gateway",
+							Name:      "test-gateway",
+							Namespace: testutil.Pointer(gatewayv1beta1.Namespace("test-gateway-ns")),
+						},
+					},
 				},
 			},
-		}
-
-		res := GetTargetRefValueFromPolicy(policy)
-		expected := "test-gateway,test-gateway-ns"
-		if res != expected {
-			t.Errorf("GetTargetRefValueFromPolicy returned %v; expected %v", res, expected)
-		}
-	})
+			want: "test-gateway,test-gateway-ns",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			expected := GetTargetRefValueFromPolicy(tt.args.policy)
+			if tt.want != expected {
+				t.Errorf("GetTargetRefValueFromPolicy returned %v; expected %v", tt.want, expected)
+			}
+		})
+	}
```

Comment:
two questions: 
1. The `_internal`  package has unit cases and they are not being executed during the `make test-unit`.  Should we run those tests? If not maybe we should get rid of them? 
2. If we are refactoring one unit test file in here, should we refactor all of them? 
And hardcoded values - especially the ones that are being used for the verification.

Reply:
Good question. I don't think we want to delete the tests, but they do indeed not run:
```
$ go test ./pkg/_internal/... -tags=unit
go: warning: "./pkg/_internal/..." matched no packages
no packages to test
```

It must be down to the build and the fact it's got this underscore in the name, if you rename it, the tests work fine:

```
$ go test -count=1 ./pkg/internal/... -tags=unit
?       github.com/Kuadrant/multicluster-gateway-controller/pkg/internal/clusterSecret  [no test files]
?       github.com/Kuadrant/multicluster-gateway-controller/pkg/internal/conditions     [no test files]
?       github.com/Kuadrant/multicluster-gateway-controller/pkg/internal/controller     [no test files]
ok      github.com/Kuadrant/multicluster-gateway-controller/pkg/internal/env    0.002s
--- FAIL: TestGracefulDelete (0.00s)
    --- FAIL: TestGracefulDelete/deleting_in_future_sets_expected_annotation (0.00s)
        gracePeriod_test.go:45: expected graceful delete error to be nil, got: grace period has not yet expired
FAIL
FAIL    github.com/Kuadrant/multicluster-gateway-controller/pkg/internal/gracePeriod    0.012s
ok      github.com/Kuadrant/multicluster-gateway-controller/pkg/internal/metadata       0.004s
ok      github.com/Kuadrant/multicluster-gateway-controller/pkg/internal/policy 0.011s
?       github.com/Kuadrant/multicluster-gateway-controller/pkg/internal/slice  [no test files]
FAIL
```
I'm not up on these internal packages, and i cant really find any info about them, but other repos don't use underscore in the name. @maleck13 You added that, what was the reason for the underscore in the package name there? Was that required?